check_point2: Interface

IgorZhiltsoff opened the pull request.

IgorZhiltsoff committed:
Created UseCase and Activity diagrams for uploading problems

IgorZhiltsoff committed:
Working on the class diagram

IgorZhiltsoff committed:
Designed upload

IgorZhiltsoff committed:
Drew up custodian

IgorZhiltsoff committed:
About to unify dumping problems and submissions

IgorZhiltsoff committed:
Tested basic custodian i/f

IgorZhiltsoff committed:
Implemented full custodian

IgorZhiltsoff committed:
Organizing

IgorZhiltsoff committed:
started switching to snake_case

IgorZhiltsoff committed:
managed UploadProblemAsTeacher structure

IgorZhiltsoff committed:
About to get rid of Custodian.extend

IgorZhiltsoff committed:
Added append to custodian

IgorZhiltsoff committed:
In the middle of smth.......

IgorZhiltsoff committed:
Implemented smth

IgorZhiltsoff committed:
Debugged upload cmd line i/f; Works now!

IgorZhiltsoff committed:
implemented running script

IgorZhiltsoff committed:
Replaced full paths with their suffixes "TPy_project/..."

IgorZhiltsoff committed:
Replaced full paths with their suffixes "TPy_project/..."

IgorZhiltsoff commented:
**TP**: last commit before DEADLINE: [c5ba60a](https://github.com/IgorZhiltsoff/TPy_project/pull/2/commits/c5ba60a78c4a0f7b1230167bc8b96cb674edfaac)

IgorZhiltsoff committed:
Implemented submission wizard!

IgorZhiltsoff committed:
Added Makefile

IgorZhiltsoff committed:
Debugged!

IgorZhiltsoff committed:
Cleaned up

IgorZhiltsoff committed:
Brushed up

IgorZhiltsoff committed:
drew up dir structure

IgorZhiltsoff committed:
drew up root

IgorZhiltsoff committed:
drew up display_readme

IgorZhiltsoff committed:
Added:
- .md support
- problem names
- Makefile
- scripts

Refreshed problem base

IgorZhiltsoff committed:
Learned how to "embed" pdfs

IgorZhiltsoff committed:
Enabled title prepend

IgorZhiltsoff committed:
Implemented and debugged display_problem_info!

IgorZhiltsoff committed:
Added return link

IgorZhiltsoff committed:
Brushed up

IgorZhiltsoff committed:
Displayed list!

IgorZhiltsoff committed:
Displayed list with refs!

IgorZhiltsoff committed:
Brushing up

IgorZhiltsoff committed:
Brushing up2

IgorZhiltsoff committed:
Brushing up2

IgorZhiltsoff committed:
Added 'back to main page' link to problem list

IgorZhiltsoff committed:
Switched to Jinja!

IgorZhiltsoff committed:
Switched to Jinja! (problem_info)

IgorZhiltsoff committed:
Got rid of unnecessary aux functions

IgorZhiltsoff committed:
Upper right corner!

IgorZhiltsoff committed:
Reduced copy&paste in templates

IgorZhiltsoff committed:
Reduced copy&paste

IgorZhiltsoff committed:
Implemented GOTO problem!

IgorZhiltsoff committed:
drew up submit.html

VTroyanGolovyan reviewed (approved):
выглядит интересно, на третьей итерации сможешь в зуме показать как оно работает?
И удали файлы с мусором пж (они все еще есть)

IgorZhiltsoff committed:
Enabled file upload!!

IgorZhiltsoff committed:
Fixed bug

IgorZhiltsoff committed:
Started implementing submit page

IgorZhiltsoff committed:
Started implementing submit page

IgorZhiltsoff committed:
Started implementing submit page

IgorZhiltsoff committed:
Improved naming in testing_protocols

IgorZhiltsoff committed:
Safely added labels

IgorZhiltsoff committed:
Got cpp tests to run
test rand failed: TestingProtocol.__init__() got an unexpected keyword argument 'language_data'
test lws failed: SKIP != AC

IgorZhiltsoff committed:
Fixed rand test

IgorZhiltsoff committed:
Passed CXX tests

IgorZhiltsoff committed:
Passed PYTHON tests

IgorZhiltsoff committed:
Banished opts from TestingProtocol class to ExecutionAndConversionData

IgorZhiltsoff committed:
Added script for running tests

IgorZhiltsoff committed:
Refactored diagrams_TestingProtocols.puml

IgorZhiltsoff committed:
Rendered

IgorZhiltsoff committed:
a

IgorZhiltsoff committed:
Revert "a"

This reverts commit 890b6e6021035de3d879f602977e9cb5116ecb97.

IgorZhiltsoff committed:
to amend

IgorZhiltsoff committed:
Enabled pre-commit hook

IgorZhiltsoff committed:
Verdict message made enum!

IgorZhiltsoff committed:
Started enforcing tempfile

IgorZhiltsoff committed:
Got rid of inout-cust copy&paste (check_with_chosen_language_data)

IgorZhiltsoff committed:
Used tempfile in randcust

IgorZhiltsoff committed:
Enforced tempfile in lws

IgorZhiltsoff committed:
Got rid of subprocess where possible in testing_protocols

IgorZhiltsoff committed:
Got rid of subprocess where possible in testing_protocols

IgorZhiltsoff committed:
Made filenames compatible with Windows

IgorZhiltsoff committed:
Prettified context manager use

IgorZhiltsoff committed:
drew up template json

IgorZhiltsoff committed:
Refactored wizard: now uploads execution_and_conversion_data_set

IgorZhiltsoff committed:
Refactored diagram

IgorZhiltsoff committed:
Attempted to refactor submission wizard's retriever module. About to test

IgorZhiltsoff committed:
Debugged submission wizard

IgorZhiltsoff committed:
Implemented process_submission for ServerIF!

IgorZhiltsoff committed:
Brushed up!

IgorZhiltsoff committed:
Enabled make once again!

IgorZhiltsoff committed:
Drew up file structure for html templates

IgorZhiltsoff committed:
Drawing up HTMLs

IgorZhiltsoff committed:
Kept drawing up HTMLs

IgorZhiltsoff committed:
Almost finished with HTMLs

IgorZhiltsoff committed:
Finished with HTMLs (at least for now...)

IgorZhiltsoff committed:
Drew up helper_upload_problem module

IgorZhiltsoff committed:
In the middle of smth...

IgorZhiltsoff committed:
In the middle of smth...2

IgorZhiltsoff committed:
Can input TL and ML floats via CMDLine!

IgorZhiltsoff committed:
Submission wizard now adds TL and ML to ExecutionAndConversionData

IgorZhiltsoff committed:
Enabled execution as nobody

IgorZhiltsoff committed:
Enforced execution as nobody in other parts of code

IgorZhiltsoff committed:
Brushed up

IgorZhiltsoff committed:
Getting ready to refactor run_code to measure resources

IgorZhiltsoff committed:
Placed resource monitoring programme
SRC: https://github.com/pshved/timeout

IgorZhiltsoff committed:
Added timeout as submodule

IgorZhiltsoff committed:
First timeout enforcement

IgorZhiltsoff committed:
Enforced timeout in rest of the places

IgorZhiltsoff committed:
Added files for testing

IgorZhiltsoff committed:
Intermediate commit
Refactoring Verdict
Refactoring run_code

IgorZhiltsoff committed:
Enforced resource consumption record!

IgorZhiltsoff committed:
Enforced resource consumption record!

IgorZhiltsoff committed:
Tested TL!

IgorZhiltsoff committed:
Removed bugs, improved naming

IgorZhiltsoff committed:
Added RE test for C++

IgorZhiltsoff committed:
Added ML and SKIP to tests!

IgorZhiltsoff committed:
Improved naming

IgorZhiltsoff committed:
Added TL test to C++

IgorZhiltsoff committed:
Added ABORT test for timelimit

IgorZhiltsoff committed:
Added missed ABORT test for checker, added ML ABORT test

IgorZhiltsoff committed:
Banished Abort tests to separate suite
Improved naming

IgorZhiltsoff committed:
Added Abort RE test
Brushed up file system

IgorZhiltsoff committed:
Added coverage (coverage is pretty good)

IgorZhiltsoff committed:
Cleaned up

IgorZhiltsoff committed:
Cleaned up 2

IgorZhiltsoff committed:
Fixed few bugs

IgorZhiltsoff committed:
Fixed few bugs

IgorZhiltsoff committed:
Rearranged files

IgorZhiltsoff committed:
Edited README.md
Started drawing up problem upload server interface

IgorZhiltsoff committed:
Added execution and conversion data to form!

IgorZhiltsoff committed:
Fully drew up htmls?

IgorZhiltsoff committed:
Implemented I/F!

IgorZhiltsoff committed:
Implemented I/F for real now

IgorZhiltsoff committed:
Drew up files

IgorZhiltsoff committed:
Drew up problem processing algorithm
Minor refactoring

IgorZhiltsoff committed:
Got smth running (but not quite)

IgorZhiltsoff committed:
Need to fix some bugs in cmd; server OK

IgorZhiltsoff committed:
Fixed bugs in CMD, successful inout upload!

IgorZhiltsoff committed:
Tested inout!

IgorZhiltsoff committed:
Added output

IgorZhiltsoff committed:
Fixed bug with permissions

IgorZhiltsoff committed:
Fixed some other bugs. Tested lws!

IgorZhiltsoff committed:
Almost done

IgorZhiltsoff committed:
Almost done! Fixing WA on incust (799)

IgorZhiltsoff committed:
Fixed incust (799), struggling with randcust (708)

IgorZhiltsoff committed:
Finished

IgorZhiltsoff committed:
Brushing up

IgorZhiltsoff committed:
Brushing up2

IgorZhiltsoff committed:
Oops! Totally forgot about client! Fixed

IgorZhiltsoff committed:
Merge pull request #3 from IgorZhiltsoff/server_interface

checkpoint_3
